### PR TITLE
Respond to cloud events validation call

### PR DESCRIPTION
**Pull request by tpaulshippy**
## Purpose
When subscribing to cloud events, Event Grid sends an OPTIONS request to verify that the subscription is acceptable. This change supports that request by responding with the proper headers as specified in the Cloud Events webhooks spec (https://github.com/cloudevents/spec/blob/v1.0.1/http-webhook.md).
* ...

## Does this introduce a breaking change?
```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
cd src/Blazor.EventGridViewer.ServerApp
dotnet run
```

* Test the code

```
cd src/Blazor.EventGridViewer.Unit.Tests
dotnet test
```

## What to Check
Verify that the following are valid
* Use a CloudEvent Event Grid Topic to subscribe to events with a webhook pointing to a hosted copy of this project. Ensure that the subscription succeeds.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# ASP.NET
```
Comment (michaeldeongreen) [inline azure-pipelines.yml:1]: @tpaulshippy, not sure if you knw but there are already pipeline files for build/test.  If you would like to add a step to publish the artifacts, the existing file can be found [here](https://github.com/Azure-Samples/eventgrid-viewer-blazor/blob/main/infrastructure/azure-pipelines/templates/steps/build-test-steps-template.yml).  If you decide to do this, please change the file to *build-test-publish-steps-template.yml*

I had planned on doing the following (feel free to do that as part of the PR):
- change trigger to main
- rename the yml files and replace master with main

File: src/Blazor.EventGridViewer.ServerApp/Controllers/EventGridController.cs [hunk 2]
```diff
@@ -38,6 +39,21 @@ public string Get()
             return "EventGridController is running...";
         }
 
+        [HttpOptions]
```
Comment (michaeldeongreen) [inline src/Blazor.EventGridViewer.ServerApp/Controllers/EventGridController.cs:42]: @tpaulshippy , can you provide some type of documentation in the README in the root directory.  Maybe the link you provided in the PR, with a brief explanation?

**Review by michaeldeongreen — CHANGES_REQUESTED** [review]
@tpaulshippy , thanks for the PR, I have left comments.

**Comment by tpaulshippy** [comment]
Going to close this as our build template changes are getting mixed in here which was not my intent.

**Pull request closed without merging**